gdal/port/cpl_conv.cpp

```diff
@@ -2007,6 +2007,111 @@ void CPL_STDCALL CPLFreeConfig()
     hConfigMutex = nullptr;
 }
 
+/************************************************************************/
+/*                    CPLLoadConfigOptionsFromFile()                    */
+/************************************************************************/
+
+/** Load configuration from a given configuration file.
+ *
+ * A configuration file is a text file in a .ini style format, that lists
+ * configuration options and their values.
+ * Lines starting with # are comment lines.
+ * 
+ * Example:
+ * <pre>
+ * [configoptions]
+ * # set BAR as the value of configuration option FOO
+ * FOO=BAR
+ * </pre>
+ *
+ * This function is typically called by CPLLoadConfigOptionsFromPredefinedFiles()
+ *
+ * @param pszFilename File where to load configuration from.
+ * @since GDAL 3.3
+ */
+void CPLLoadConfigOptionsFromFile(const char* pszFilename)
+{
+    VSILFILE* fp = VSIFOpenL(pszFilename, "rb");
+    if( fp == nullptr )
+        return;
+    CPLDebug("CPL", "Loading configuration from %s", pszFilename);
+    const char* pszLine;
+    bool bInConfigOptions = false;
+    while( (pszLine = CPLReadLine2L(fp, -1, nullptr)) != nullptr )
+    {
+        if( pszLine[0] == '#' )
+        {
+            // Comment line
+        }
+        else if( strcmp(pszLine, "[configoptions]") == 0 )
+        {
+            bInConfigOptions = true;
+        }
```

Comment:
To make the file extendable, add something like:
```
else if (STARTS_WITH(pszLine, "["))
{
  bInConfigOptions = false;
}
```

Reply:
Done